### PR TITLE
feat: add rust binding for baby bear row major matrix

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR introduces c APIs and rust bindings for `RowMajorMatrix<BabyBear>`

### CODE REVIEW

**Review by GideokKim — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

File: tachyon/c/math/matrix/baby_bear_row_major_matrix.h [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/**
+ * @file baby_bear_row_major_matrix.h
+ * @brief Row major matrix operations for the baby bear.
```
Comment (ashjeong) [inline tachyon/c/math/matrix/baby_bear_row_major_matrix.h:3]: Please change all instances to either "for the baby bear field" or "for baby bear"

File: tachyon/c/math/matrix/baby_bear_row_major_matrix.h [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * This header file defines the structure and API for row major matrix for the
+ * baby bear This includes creation, cloning, and destruction of row major
```
Comment (ashjeong) [inline tachyon/c/math/matrix/baby_bear_row_major_matrix.h:6]:
```suggestion
 * baby bear. This includes creation, cloning, and destruction of row major

File: tachyon/c/math/matrix/baby_bear_row_major_matrix.h [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ * @brief Creates a new instance of a row major matrix.
+ *
+ * Allocates and initializes a new row major matrix structure for use
+ * in cryptographic algorithms.
+ *
+ * @param ptr A pointer to the matrix data.
+ * @param rows The number of rows.
+ * @param cols The number of column.
```
Comment (ashjeong) [inline tachyon/c/math/matrix/baby_bear_row_major_matrix.h:40]:
```suggestion
 * @param cols The number of columns.

File: tachyon/c/math/matrix/baby_bear_row_major_matrix.h [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+ * @return The element of the row major matrix at a given row and column index.
+ */
+TACHYON_C_EXPORT tachyon_baby_bear
+tachyon_baby_bear_row_major_matrix_get_element(
+    const tachyon_baby_bear_row_major_matrix* matrix, size_t row, size_t col);
+
+/**
+ * @brief Returns a const pointer to the matrix data.
+ *
+ * @param matrix A pointer to the matrix.
```
Comment (ashjeong) [inline tachyon/c/math/matrix/baby_bear_row_major_matrix.h:103]: This suggestion is added on to standardize with the return value statement right after, yet if this change is made, all similar instances should be changed as well! (such as on line 91)
```suggestion
 * @param matrix A const pointer to the matrix.

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**